### ISSUE TITLE
Build macro example that breaks display

### ISSUE THREAD

**Issue by RblSb** [HaxeFoundation/haxe#11211]
This example shows error if `@:build` macro is commented out, and prints errors to output when you hover on most var names, if there is a `@:build`.
Minimized from Safety library, so basically this happen with any code with safety macro operator, and with any error without `Common.ignore_error` wrap.
Not sure if this is not duplicate, or best solution is just to wrap everything in `ignore_error`, but there is a code.
```haxe
@:build(SafeAst.build())
class Main {
	static function main() {
		var errRange = 0;
		final prevId = trace("arg");
		if (errRange != null) {
			final hasComma = false;
		}
	}
}

SafeAst.hx:

import haxe.macro.Context;
import haxe.macro.Expr;
using haxe.macro.Tools;

class SafeAst {
	macro public static function build():Array<Field> {
		final fields = Context.getBuildFields();
		for (field in fields) {
			var expr:Null<Expr> = switch(field.kind) {
				case FVar(_, e): e;
				case FProp(_, _, _, e): e;
				case FFun(fn): fn.expr;
			}
			if(expr != null) {
				expr.expr = transform(expr).expr;
			}
		}
		return fields;
	}

	static function transform(expr:Expr):Expr {
		// !. operator example
		// switch expr {
		// 	case macro $target!.$field:
		// 		expr = macro @:pos(expr.pos) {
		// 			var _v_ = $target;
		// 			_v_ == null ? null : _v_.$field;
		// 		};
		// 		return expr;
		// 	case _:
		// }
		return expr.map(transform);
	}
}

[Error - 08:31:15] Request textDocument/hover failed.
  Message: Variables of type Void are not allowed
  Code: -32603 
```
[a_macro_breaks_display.zip](https://github.com/HaxeFoundation/haxe/files/11428796/a_macro_breaks_display.zip)

**Comment by kLabz** [comment]
Not sure what you're expecting there because this errors on normal compilation too, without compilation server.
Hover also seems fine locally after some quick tests.

Edit:
Ok I can repro some hover issues on 4.3.x, but not on nightlies.

**Comment by kLabz** [comment]
Will update 4.3_bugfix with what is expected to get there and see if this still reproduces there.
If it does, there might be something more we want to add to 4.3_bugfix.
If it doesn't, I guess we can close this.

**Comment by kLabz** [comment]
Strangely this was fixed by #11163, which if added to 4.3_bugfix makes hover requests behave here.

I kinda wanted to add that to 4.3_bugfix, but it can be a bit problematic. Wdyt @Simn ?

**Comment by Simn** [comment]
Yes I thought that was the idea anyway.

**Comment by kLabz** [comment]
Fixed in nightlies and [`4.3_bugfix`](https://github.com/HaxeFoundation/haxe/tree/4.3_bugfix)
